src/dual.jl

```diff
@@ -360,6 +360,9 @@ end
 @inline Base.one(d::Dual) = one(typeof(d))
 @inline Base.one(::Type{Dual{T,V,N}}) where {T,V,N} = Dual{T}(one(V), zero(Partials{N,V}))
 
+@inline Base.oneunit(d::Dual) = oneunit(typeof(d))
```

Comment:
I'm not sure if there are any examples of numeric types where `oneunit` is only defined for instances but not for types (could imagine unitful numbers where the unit is not a type parameter). In such cases falling back on the definition for `Dual` here would fail. Therefore I think the more correct implementation would be
```suggestion
@inline Base.oneunit(d::Dual{T}) where {T} = Dual{T}(oneunit(primal(d)), zero(partials(d)))
```
(IMO the same reasoning applies also to other functions already defined in dual.jl)

Reply:
Fair enough

Reply:
There is exactly this in DynamicQuantities.jl:
https://github.com/SymbolicML/DynamicQuantities.jl/blob/1d62cec8a7df1867b240f8c5d9183063c0a00a41/src/utils.jl#L107

Reply:
Hmm, the suggestion fails with `UndervarError: primal not defined`, maybe is `value` instead of `primal`?

Reply:
Ah sorry, you're probably right, I guess I misremembered the function name.